### PR TITLE
Upgrade to IPFS 0.27

### PR DESCRIPTION
**Pull request by pgte**
- [x] Use websocket-star (instead of webrtc-star) transport
- [x] Migrate repo version
- [x] Test
  - [x] Presence
  - [x] Shared editing
  - [x] Reconnect

### CODE REVIEW

**Comment by pgte** [comment]
Currently running into this issue: https://github.com/libp2p/js-libp2p-websocket-star/issues/34

File: src/backend/ipfs.js [hunk 1]
```diff
@@ -29,14 +27,14 @@ function maybeCreateIPFS (_ipfs) {
       config: {
         Addresses: {
           Swarm: [
-            '/dns4/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star'
+            '/dns4/ws-star-signal-1.servep2p.com/tcp/443/wss/p2p-websocket-star'
```
Comment (daviddias) [inline src/backend/ipfs.js]: Why the change??

**Review by daviddias — CHANGES_REQUESTED** [review]

File: src/backend/ipfs.js (hunk 1 above)
Comment (pgte) [reply src/backend/ipfs.js]: Switched because this was the recommended rendez-vous server address on the docs..

Comment (pgte) [reply src/backend/ipfs.js]: We should eventually switch to a peerpad-specific one..

Comment (daviddias) [reply src/backend/ipfs.js]: Good point, missed updating there https://github.com/libp2p/js-libp2p-websocket-star/pull/40

Comment (daviddias) [reply src/backend/ipfs.js]: I believed people would use https://github.com/ipfs/js-ipfs#is-there-a-more-stable-alternative-to-webrtc-star-that-offers-a-similar-functionality for reference

**Review by daviddias — APPROVED** [review]

**Comment by daviddias** [comment]
@pgte can you break this PR into two?
- Upgrade to js-ipfs 0.27 (continue to use webrtc-star)
- Move away from webrtc-star and use websocket-star

There are some significant perf improvements in 0.27 that you might want to start using right away

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "bootstrap": "^3.3.7",
     "bs58": "^4.0.1",
     "history": "^4.7.2",
-    "ipfs": "^0.26.0",
+    "ipfs": "^0.27.0",
```
Comment (daviddias) [inline package.json]: Better change this to `~ `

**Review by daviddias — CHANGES_REQUESTED** [review]

File: src/backend/index.js [hunk 3]
```diff
@@ -25,7 +22,11 @@ class Backend extends EventEmitter {
     this.keys = {
       generateSymmetrical: generateSymmetricalKey
     }
+<<<<<<< HEAD
     this.network = new Network(this.room)
+=======
+    this._handleError = this._handleError.bind(this)
+>>>>>>> error handling and upgraded to IPFS 0.27 files API
```
Comment (daviddias) [inline src/backend/index.js]: merge conflict

**Review by daviddias — CHANGES_REQUESTED** [review]

**Pull request merged**